tools/walletextension/api/routes.go

```diff
@@ -216,3 +237,249 @@ func submitViewingKeyRequestHandler(walletExt *walletextension.WalletExtension,
 		return
 	}
 }
+
+func joinRequestHandler(walletExt *walletextension.WalletExtension, userConn userconn.UserConn) {
+	// todo (@ziga) add protection against DDOS attacks
+	_, err := userConn.ReadRequest()
+	if err != nil {
+		return
+	}
+
+	// generate new key-pair
+	viewingKeyPrivate, err := crypto.GenerateKey()
+	viewingPrivateKeyEcies := ecies.ImportECDSA(viewingKeyPrivate)
+	if err != nil {
+		userConn.HandleError(fmt.Sprintf("could not generate new keypair: %s", err))
+		return
+	}
+	viewingPublicKeyBytes := crypto.CompressPubkey(&viewingKeyPrivate.PublicKey)
+
+	// create UserID and store it in the database with the private key
+	userID := crypto.Keccak256Hash(viewingPublicKeyBytes)
+	err = walletExt.Storage.AddUser(userID.Bytes(), crypto.FromECDSA(viewingPrivateKeyEcies.ExportECDSA()))
+	if err != nil {
+		userConn.HandleError("Internal error")
+		walletExt.Logger().Error(fmt.Sprintf("failed to save user to the database: %s", err))
+		walletExt.Logger().Error(fmt.Sprintf("failed to save user to the database: %s", err))
+		return
+	}
```

Comment:
The route should request the WE to handle the VK gen. The route doesn't need to know about vk storage, etc.

Reply:
moved to WE